### ISSUE TITLE
Comments not displayed for `typedef` and `macros`

### ISSUE THREAD

**Issue by yoavrv** [microsoft/vscode-cpptools#8320]
Issue Type: <b>Bug</b>

Any comment at the start of the file seems to overrides the tooltips for typedefs and macros

for example file.c contains:
```c
/* vim: set ts=4 sts=4 sw=4 noet : */
#include <stdio.h>
int no_description;
int my_global; //this is a global variable
typedef unsigned int my_uint; // this is my special uint
// this is some magic number
#define MAGIC_NUMBER 512
```
when hovering over `my_global`, the tooltip shows "this is my global variable", but hovering over `my_uint` or `MAGIC_NUMBER` shows " vim: set ts=4 sts=4 sw=4 noet : ".

without the comment at the top, `my_uint`  and MAGIC_NUMBER shows nothing (which I guess is the correct behavior?).

I'm not sure if it's helpful to know, but when undoing/redoing the top comment rapidly, `no_description`'s tooltip would occasionly show " vim: set ts=4 sts=4 sw=4 noet : " before reverting to nothing, and `my_uint`'s would show "this is my uint" before reverting to nothing

Edit: I don't know vim, this is just the comment that the original file came with that cause the problem

Extension version: 1.7.1
VS Code version: Code 1.61.2 (6cba118ac49a1b88332f312a8f67186f7f3c1643, 2021-10-19T14:57:20.575Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No
Remote OS version: Linux x64 4.19.128-microsoft-standard
Remote OS version: Linux x64 4.19.128-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz (8 x 1800)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.82GB (1.58GB free)|
|Process Argv|--folder-uri=vscode-remote://wsl+Ubuntu-20.04/opt/workspace/msc_project/cluster-trisurf --remote=wsl+Ubuntu-20.04 --crash-reporter-id 6b52cd83-0bde-4b1f-ba92-9b01751b74a4|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-20.04|
|OS|Linux x64 4.19.128-microsoft-standard|
|CPUs|Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz (8 x 1800)|
|Memory (System)|6.05GB (5.19GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-20.04|
|OS|Linux x64 4.19.128-microsoft-standard|
|CPUs|Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz (8 x 1800)|
|Memory (System)|6.05GB (5.19GB free)|
|VM|0%|
</details>

**Comment by tim-bargen** [comment]
I've noticed this issue as well. It seems to only affect typedefs for primitive data types; structs, enums etc. all have the correct comments show up in the tooltip.

**Comment by sean-mcmanus** [comment]
Fixed with 1.11.3 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.11.3
